compiler/include/expr.h

```diff
@@ -165,32 +165,51 @@ class UnresolvedSymExpr : public Expr {
 // ContextCallExpr. Therefore, it is important to use toCallExpr()
 // instead of casting to CallExpr* directly.
 class CallExpr : public Expr {
- public:
-  Expr* baseExpr;         // function expression
-  AList argList;          // function actuals
-  PrimitiveOp* primitive; // primitive expression (baseExpr == NULL)
-
-  bool partialTag;
-  bool methodTag; ///< Set to true if the call is a method call.
-  // It is used in gatherCandidates to filter out method field extraction
-  // (partials).
-  // TODO: Maybe use a new primitive to represent partials, and get rid of this tag.
-  bool square; // true if call made with square brackets
-
-  CallExpr(BaseAST* base, BaseAST* arg1 = NULL, BaseAST* arg2 = NULL,
-           BaseAST* arg3 = NULL, BaseAST* arg4 = NULL, BaseAST* arg5 = NULL);
-  CallExpr(PrimitiveOp *prim, BaseAST* arg1 = NULL, BaseAST* arg2 = NULL,
-           BaseAST* arg3 = NULL, BaseAST* arg4 = NULL, BaseAST* arg5 = NULL);
-  CallExpr(PrimitiveTag prim, BaseAST* arg1 = NULL, BaseAST* arg2 = NULL,
-           BaseAST* arg3 = NULL, BaseAST* arg4 = NULL, BaseAST* arg5 = NULL);
-  CallExpr(const char* name, BaseAST* arg1 = NULL, BaseAST* arg2 = NULL,
-           BaseAST* arg3 = NULL, BaseAST* arg4 = NULL, BaseAST* arg5 = NULL);
+public:
+  PrimitiveOp* primitive;        // primitive expression (baseExpr == NULL)
+  Expr*        baseExpr;         // function expression
+
+  AList        argList;          // function actuals
+
+  bool         partialTag;
+  bool         methodTag;        // Set to true if the call is a method call.
```

Comment:
I find myself wishing all too often for comments here explaining partialTag and methodTag. I understand this is not the goal of this PR. Since last time I wished for this, I came to understand that these are written in this way partly to resolve an ambiguity between
   bar.baz()
It could be that bar is a parentheses-less method returning something with a .baz() method; or it could be that bar is an object upon which a .baz() method can be invoked. It might be valuable to merely mention this situation.